node/src/main.rs

```diff
@@ -2,11 +2,15 @@ use clap::Parser;
 use std::error::Error;
 use std::net::ToSocketAddrs;
 use tokio::net::{TcpListener, TcpStream};
+use tokio::sync::mpsc;
 use tokio_util::codec::{FramedRead, FramedWrite, LengthDelimitedCodec};
 
 mod cli;
 mod connection;
 mod protocol;
+mod template;
+
+use crate::template::{template_consumer, template_poll};
```

Comment:
The prefix "template" is being repeated here.

`use crate::block_template::{consumer, poll}` is cleaner.

Also, I prefer the following way to import and use those imports.

```
use crate::block_template
...
tokio::spawn(block_template::poll)
```